### PR TITLE
mysterious-mice

### PR DESCRIPTION
**Pull request by onatoko**
Pull request for Python Discord Summer Code Jam 2020

### CODE REVIEW

**Comment by SebastiaanZ** [comment]
No merge issues.

No flake8 linting errors

File: mysterious-mice/spacebook/posts/views.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    """
+
+    model = ImagePost
+    template_name = "posts/home.html"
+    context_object_name = "posts"
+    ordering = ["-date"]
+    paginate_by = 16
+
+    # if a valid username provided, only return that rover's posts
+    def get_queryset(self):
```
Comment (eivl) [inline mysterious-mice/spacebook/posts/views.py:21]: nice exception handeling. looks very pythonic. do imagepost need to be inside the try as well?

File: mysterious-mice/spacebook/posts/views.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        username = self.request.GET.get("username")
+        try:
+            rover = Rover.objects.get(username=username)
+            imageposts = ImagePost.objects.filter(rover=rover).order_by("-date")
+        except ObjectDoesNotExist:
+            imageposts = ImagePost.objects.all().order_by("-date")
+        return imageposts
+
+    # pass additional context if only displaying posts from one rover
+    def get_context_data(self, **kwargs):
```
Comment (eivl) [inline mysterious-mice/spacebook/posts/views.py:31]: well made context handling.

File: mysterious-mice/spacebook/posts/views.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+            username_get = f"?username={username}"
+            context["username"] = username
+        except ObjectDoesNotExist:
+            # if no valid username, navigate to next/prev image regardless of rover
+            imageposts = list(ImagePost.objects.all().order_by("-date"))
+            username_get = ""
+
+        # if only one image returned, no links required
+        prev_image_pk = next_image_pk = "#"
+        if len(imageposts) > 1:
```
Comment (eivl) [inline mysterious-mice/spacebook/posts/views.py:76]: looks a bit messy with all the ifs and else's. refactor it out might make it easier on the eyes.

**Comment by eivl** [comment]
Crash landing! did not work for me. is that not intended?

**Comment by onatoko** [comment]
> Crash landing! did not work for me. is that not intended?
> @onatoko ?

@eivl I'm coming across the same issue now, and it looks like the weekly weather page is also not working. I believe there's something that we're getting from NASA's Mars Weather API that we hadn't accounted for and is causing the issue. There's a quick change I can make to the game file to make it playable if that would be acceptable. A few lines that refer to the weather API need to be commented out, then two variables need to be defined. I can create a new pull request if that would be acceptable

**Comment by onatoko** [comment]
@eivl I have created a pull request with the fixes, hopefully the game should be playable now (it's working again on my end). If you come across anything else feel free to let us know. Thanks!

**Pull request merged**